apps/admin/pages/api/auth/social-auth/[provider].ts

```diff
@@ -0,0 +1,73 @@
+import { SessionUser, Session } from '@wsvvrijheid/types'
+import { fetcher, mutation, request, sessionOptions } from '@wsvvrijheid/utils'
+import { withIronSessionApiRoute } from 'iron-session/next'
+import { NextApiResponse, NextApiRequest } from 'next'
+
+const route = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { provider } = req.query
+  if (req.method === 'POST') {
+    try {
+      let url = `/api/auth/${provider}/callback?access_token=${req.body.access_token}`
+
+      if (provider === 'twitter') {
+        url = `/api/auth/${provider}/callback?access_token=${req.body.access_token}&access_secret=${req.body.access_secret}`
+      }
+
+      const socialLoginResponse = await fetcher(req.body.access_token).get(url)
+
+      const token = socialLoginResponse.data.jwt
+      const userId = socialLoginResponse.data.user?.id
+
+      if (token) {
+        const artistResponse = await request(token)({
+          url: 'api/artists',
+          filters: { user: { id: { $eq: userId } } },
+        })
+
+        if (!artistResponse?.data?.[0]) {
+          await mutation(token).post('api/artists', {
+            data: {
+              user: userId,
+              name: socialLoginResponse.data.user?.username,
+            },
+          })
+        }
+
+        const response = await request(token)({
+          url: 'api/users',
+          filters: { id: { $eq: userId } },
+        })
```

Comment:
Probably it makes more sense to fetch the user by id.

```suggestion
        const response = await request(token)<User>({
          url: `api/users/${userId}`,
        })
```